sematic/resolvers/silent_resolver.py

```diff
@@ -16,18 +17,44 @@ class SilentResolver(StateMachineResolver):
     def _schedule_future(self, future: AbstractFuture) -> None:
         self._run_inline(future)
 
+    def _activate_resources(self, future):
+        updated_resources = []
+        for resource in future.props.external_resources:
+            resource = resource.activate(is_local=True)
+            while not (
+                resource.status.state == ResourceState.ACTIVE
+                or resource.status.state.is_terminal()
+            ):
+                resource = resource.update()
```

Comment:
Is this not going to hammer whatever `.update` is doing (e.g. checking on a resource)?